redis-sink/src/main/java/org/springframework/cloud/stream/module/redis/CustomRedisConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package org.springframework.cloud.stream.module.redis;
+
+import java.net.UnknownHostException;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import redis.clients.jedis.Jedis;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.redis.RedisAutoConfiguration;
+import org.springframework.boot.autoconfigure.redis.RedisProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.data.redis.core.RedisOperations;
+
+/**
+ * Creates a dedicated RedisConnectionFactory (that is, different from the spring-cloud-stream one).
+ *
+ * @author Eric Bottard
+ */
+@Configuration
+@ConditionalOnClass({ JedisConnection.class, RedisOperations.class, Jedis.class })
+@EnableConfigurationProperties
+public class CustomRedisConfiguration extends RedisAutoConfiguration {
```

Comment:
But it should be - it only works now because there is a component scan on that package, but if we wanted to reuse this module in another module, it would have to know to component scan that package...  I'm going to take a shot at rewriting this with some changes and post it to my own branch.